stop_and_wait_plugin/test/test_stopandwait_plugin.cpp

```diff
@@ -80,62 +82,49 @@ TEST(StopandWait, TestStopandWaitPlanning)
+  req->header.stamp = rclcpp::Time(0.0* 1e9);
 
-  cav_msgs::Maneuver maneuver;
-  maneuver.type = cav_msgs::Maneuver::STOP_AND_WAIT;
+  carma_planning_msgs::msg::Maneuver maneuver;
+  maneuver.type = carma_planning_msgs::msg::Maneuver::STOP_AND_WAIT;
   maneuver.stop_and_wait_maneuver.start_dist = 5.0;
-  maneuver.stop_and_wait_maneuver.start_time = ros::Time(0.0);
+  maneuver.stop_and_wait_maneuver.start_time = rclcpp::Time(0.0* 1e9);
   maneuver.stop_and_wait_maneuver.start_speed = 8.9408;
 
   maneuver.stop_and_wait_maneuver.end_dist = 55;
-  maneuver.stop_and_wait_maneuver.end_time = ros::Time(11.175999999999998);
+  maneuver.stop_and_wait_maneuver.end_time = rclcpp::Time(11.175999999999998* 1e9);
 
-  req.maneuver_plan.maneuvers.push_back(maneuver);
-  req.maneuver_index_to_plan = 0;
+  req->maneuver_plan.maneuvers.push_back(maneuver);
+  req->maneuver_index_to_plan = 0;
 
-  cav_srvs::PlanTrajectoryResponse resp;
+  carma_planning_msgs::srv::PlanTrajectory::Response::SharedPtr resp;
   plugin.plan_trajectory_cb(req, resp);
 
   double dist = 0;
-  double vel = resp.trajectory_plan.initial_longitudinal_velocity;
+  double vel = resp->trajectory_plan.initial_longitudinal_velocity;
   bool first= true;
   lanelet::BasicPoint2d prev_point;
-  ros::Time prev_time = ros::Time(0.0);
-  for (auto point : resp.trajectory_plan.trajectory_points) {
+  rclcpp::Time prev_time = rclcpp::Time(0.0* 1e9);
+  for (auto point : resp->trajectory_plan.trajectory_points) {
     lanelet::BasicPoint2d p(point.x, point.y);
-    ROS_INFO_STREAM("Y: " << point.y);
+    RCLCPP_INFO_STREAM(nh->get_logger(),"Y: " << point.y);
     double delta = 0;
     if (first) {
       first = false;
     } else {
       delta = lanelet::geometry::distance2d(p, prev_point);
       dist += delta;
-      ROS_INFO_STREAM("delta: " << delta << " timediff: " << (point.target_time - prev_time).toSec() << "pre_vel: " << vel);
-      vel = (2.0 * delta / (point.target_time - prev_time).toSec()) - vel;
+      RCLCPP_INFO_STREAM(nh->get_logger(),"delta: " << delta << " timediff: " << (rclcpp::Time(point.target_time) - prev_time).seconds() << "pre_vel: " << vel);
+      vel = (2.0 * delta / (rclcpp::Time(point.target_time) - prev_time).seconds()) - vel;
     }
-    ROS_ERROR_STREAM("point time: " << point.target_time.toSec() << " dist: " << dist << " vel: " << vel);
+    RCLCPP_ERROR_STREAM(nh->get_logger(),"point time: " << rclcpp::Time(point.target_time).seconds() << " dist: " << dist << " vel: " << vel);
     prev_point = p;
     prev_time = point.target_time;
   }
 }
 }  // namespace stop_and_wait_plugin
 
-// Run all the tests
-int main(int argc, char** argv)
-{
-  testing::InitGoogleTest(&argc, argv);
-  ros::Time::init();
-  ROSCONSOLE_AUTOINIT;
-  if( ros::console::set_logger_level(ROSCONSOLE_DEFAULT_NAME, ros::console::levels::Debug) ) { // Change to Debug to enable debug logs
-    ros::console::notifyLoggerLevelsChanged();
-  }
-  auto res = RUN_ALL_TESTS();
-  
-  return res;
-}
```

Comment:
Please include ROS 2 version of unit test main. Any other ros2 package will have it like plan_delegator, ILC etc